cmd/offline/offline.go

```diff
@@ -22,7 +22,7 @@ import (
 // tried sending them to the API. If we tried sending to API already,
 // to the API. Used when we have heartbeats unsent to API.
 func SaveHeartbeats(v *viper.Viper, heartbeats []heartbeat.Heartbeat, queueFilepath string) error {
-	params, err := loadParams(v, heartbeats == nil)
```

Comment:
Don't we want to skip re-loading the param heartbeat when only loading params for saving extra heartbeats? Otherwise we'll accidentally save the param heartbeat to offline db when we only wanted to save the extra heartbeats over the max.

https://github.com/wakatime/wakatime-cli/pull/640#discussion_r799632901

Reply:
We have a guard few lines ahead [here](https://github.com/wakatime/wakatime-cli/blob/bugfix/offline-sanitization/cmd/offline/offline.go#L38). That way we ensure to build heartbeats (_including the main one_) only when it comes null. We need to always load `HeartbeatParam` because all sanitization rules are in it.